src/main/java/seedu/address/model/delivery/DeliveryStatusPredicate.java

```diff
@@ -0,0 +1,36 @@
+package seedu.address.model.delivery;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests if a {@code Delivery}'s status matches the given status.
+ */
+public class DeliveryStatusPredicate implements Predicate<Delivery> {
```

Comment:
Should we reuse the DeliveryStatusMatchInputPredicate class instead for this predicate since it is also checking if two deliveries have matching status

Reply:
Yep, we shall use the existing DeliveryStatusMatchInputPredicate class